hivemind/p2p/p2p_daemon.py

```diff
@@ -146,27 +150,41 @@ async def send_raw_data(byte_str, writer):
         writer.write(request)
 
     @staticmethod
-    async def send_data(data, writer):
+    async def send_message_pack(data, writer):
         raw_data = MSGPackSerializer.dumps(data)
         await P2P.send_raw_data(raw_data, writer)
 
     @staticmethod
     async def send_protobuf(protobuf, out_proto_type, writer):
         if type(protobuf) != out_proto_type:
-            error = TypeError('Unary handler returned protobuf of wrong type.')
-            await P2P.send_raw_data(MSGPackSerializer.dumps(error), writer)
-            raise error
+            raise TypeError('Unary handler returned protobuf of wrong type.')
         await P2P.send_raw_data(protobuf.SerializeToString(), writer)
 
     @staticmethod
-    async def receive_raw_data(reader: asyncio.StreamReader):
-        header = await reader.readexactly(P2P.HEADER_LEN)
+    async def send_protobuf_with_error(protobuf, out_proto_type, writer):
```

Comment:
```suggestion
    async def send_protobuf_or_error(protobuf, out_proto_type, writer):
```

The current name may be misleading since the protobuf does not always contain an error.